### PR TITLE
Rewrite while into for, take opportunities to save semi-colons

### PR DESCRIPTION
**Pull request by eldritchconundrum**

### CODE REVIEW

File: Checker/compression_test.fs [hunk 1]
```diff
@@ -31,6 +31,8 @@ let testFiles = [
     "buoy.frag"
     "orchard.frag"
     "robin.frag"
+    "sult.frag"
+    "../unit/loop.frag"
```
Comment (laurentlb) [inline Checker/compression_test.fs:35]: I'd rather avoid synthetic tests in this list.

File: tests/real/from-the-seas-to-the-stars.frag.expected [hunk 2]
```diff
@@ -432,8 +432,7 @@ void main()
                       w.y=R();
                       w.z=R();
                       w.w=R();
-                      if(w.w<.5)
-                        ;
+                      w.w<.5;
```
Comment (laurentlb) [inline tests/real/from-the-seas-to-the-stars.frag.expected:435]: This is something we could simplify later

**Review by laurentlb — APPROVED** [review]

File: tests/commands.txt [hunk 3]
```diff
@@ -22,6 +22,7 @@
 --no-renaming --no-inlining --format indented -o tests/unit/vectors.frag.expected tests/unit/vectors.frag
 --no-renaming --format indented -o tests/unit/deadcode.frag.expected tests/unit/deadcode.frag
 --no-renaming --no-inlining --format indented -o tests/unit/augmented.frag.expected tests/unit/augmented.frag
+--no-remove-unused -o tests/unit/loop.frag.expected tests/unit/loop.frag
```
Comment (laurentlb) [inline tests/commands.txt:25]: you can disable renaming here

**Comment by laurentlb** [comment]
This could be added to the README

File: Checker/compression_test.fs (hunk 1 above)
Comment (eldritchconundrum) [reply Checker/compression_test.fs:35]: Me too. But I wanted to know if this rewrite also reduces compressed size. We need to grab more real test shaders.

**Pull request merged**